Document/0x05b-Basic-Security_Testing.md

```diff
@@ -472,16 +477,47 @@ $ frida-ps -U
   276  adbd
   956  android.process.media
   198  bridgemgrd
- 1191  com.android.nfc
- 1236  com.android.phone
- 5353  com.android.settings
-  936  com.android.systemui
+30692  com.android.chrome
+30774  com.android.chrome:privileged_process0
+30747  com.android.chrome:sandboxed
+30834  com.android.chrome:sandboxed
+ 3059  com.android.nfc
+ 1526  com.android.phone
+17104  com.android.settings
+ 1302  com.android.systemui
 (...)
 ```
 
-The -U option lets Frida search for USB devices or emulators.
+Or restrict the list with the `-Uai` flag combination to get all apps (`-a`) currently installed (`-i`) on the connected USB device (`-U`):
 
-Use `frida CLI` to work with Frida interactively. It hooks into a process and gives you a command line interface to Frida's API.
+```bash
+$ frida-ps -Uai
+  PID  Name                                      Identifier
+-----  ----------------------------------------  ---------------------------------------
+  766  Android System                            android
+30692  Chrome                                    com.android.chrome
+ 3520  Contacts Storage                          com.android.providers.contacts  
+    -  Uncrackable1                              sg.vantagepoint.uncrackable1
+    -  drozer Agent                              com.mwr.dz
+```
+
+This will show the names and identifiers of all apps, if they are currently running it will also show their PIDs. Search for your app in the list and take a note of the PID or its name/identifier. From now on you'll refer to your app by using one of them. A recommendation is to use the identifiers, as the PIDs will change on each run of the app. For example let's take `com.android.chrome`. You can use this string now on all Frida tools, e.g. on the Frida CLI, on frida-trace or from a Python script.
+
+###### Tracing Native Libraries with frida-trace
+
+To trace specific (low-level) library calls, you can use the `frida-trace` command line tool:
+
+```shell
+$ frida-trace -U com.android.chrome -i "open"
+```
+
+This generates a little JavaScript in `__handlers__/libc.so/open.js`, which Frida injects into the process. The script traces all calls to the `open` function in `libc.so`. You can modify the generated script according to your needs with Frida [JavaScript API](https://www.frida.re/docs/javascript-api/).
+
+Unfortunately tracing high-level methods of Java classes is not yet supported (but might be [in the future](https://github.com/frida/frida-python/issues/70 "Support for tracing high-level methods of Java Classes via patterns")).
+
+###### Frida CLI and the Java API
+
+Use the Frida CLI tool (`frida`) to work with Frida interactively. It hooks into a process and gives you a command line interface to Frida's API.
```

Comment:
this seems to be the first time you introduce the Frida CLI tool with just (`frida`) can you move that introduction to the top of this chapter?

Reply:
I guess you mean of this section? I put that now in Frida's intro for Android, when we first talk about these tools. Frida CLI (`frida`)was also introduced in 4c

Remember that on Android, you can also benefit from the built-in tools provided when installing Frida, that includes the Frida CLI (`frida`), `frida-ps`, `frida-ls-devices` and `frida-trace`, to name some of them.